File: E4/E04-01/app/src/main/java/com/example/e04_01/activities/MainActivity.java
package com.example.e04_01.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import com.example.e04_01.R;
import com.example.e04_01.adapters.EmployeesAdapter;
import com.example.e04_01.enums.Gender;
import com.example.e04_01.enums.Role;
import com.example.e04_01.viewmodels.Employee;
import com.google.firebase.auth.FirebaseAuth;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


    }
}

File: 2022-SOPM/E05/E05-02/app/src/main/java/com/example/recyclerviewexercise/MainActivity.kt
package com.example.recyclerviewexercise

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.recyclerviewexercise.adapters.StudentsAdapter
import com.example.recyclerviewexercise.enums.Gender
import com.example.recyclerviewexercise.viewmodels.StudentViewModel

class MainActivity : AppCompatActivity() {
    val TAG = "MainActivity"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val studentsAdapter = StudentsAdapter()

        val recyclerView = findViewById<RecyclerView>(R.id.recycler_view)
        recyclerView.adapter = studentsAdapter
        recyclerView.layoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
        val students = ArrayList<StudentViewModel>()
        students.add(StudentViewModel("Armend", "Ukehaxhaj", Gender.Male))
        students.add(StudentViewModel("Armend1", "Ukehaxhaj1", Gender.Male))
        students.add(StudentViewModel("Armend2", "Ukehaxhaj2", Gender.Male))
        students.add(StudentViewModel("Armend3", "Ukehaxhaj3", Gender.Male))
        students.add(StudentViewModel("Armend4", "Ukehaxhaj4", Gender.Male))
        students.add(StudentViewModel("Armend4", "Ukehaxhaj4", Gender.Male))
        students.add(StudentViewModel("Armend4", "Ukehaxhaj4", Gender.Male))
        students.add(StudentViewModel("Armend4", "Ukehaxhaj4", Gender.Male))
        students.add(StudentViewModel("Armend4", "Ukehaxhaj4", Gender.Male))
        students.add(StudentViewModel("Armend4", "Ukehaxhaj4", Gender.Male))
        students.add(StudentViewModel("Armend4", "Ukehaxhaj4", Gender.Male))
        students.add(StudentViewModel("Armend4", "Ukehaxhaj4", Gender.Male))
        students.add(StudentViewModel("Armend4", "Ukehaxhaj4", Gender.Male))
        students.add(StudentViewModel("Armend4", "Ukehaxhaj4", Gender.Male))
        students.add(StudentViewModel("Armend4", "Ukehaxhaj4", Gender.Male))
        studentsAdapter.setData(students)
    }
}